### ISSUE TITLE
Tutor nightly polluted by olive-specific changes

### ISSUE THREAD

**Issue by ARMBouhali** [overhangio/tutor#767]
**TL;DR -** Tutor 15.1.0-nightly runs Open edX on `olive.1` release instead of `master` branch

**Bug description**
1. Tutor nightly branch is supposed to use the `master` branch, but it has been updated with `open-release/olive.1`.
2. Commits backported from `edx-platform/master` branch shouldn't be included in the `openedx` image's Dockerfile.

**How to reproduce**
1. Install tutor nightly
2. Run `tutor config save --set OPENEDX_COMMON_VERSION=master` to point to the master branch for all openedx repos.
3. Run `tutor images build openedx` and it should fail to apply the patch and output the following log

```
Step 24/106 : RUN curl -fsSL https://github.com/openedx/edx-platform/commit/20b93b8b01276edadddfbbb67f15714fddd81c31.patch | git am
 ---> Running in be498efa61f9
error: patch failed: xmodule/templates/problem/circuitschematic.yaml:65
error: xmodule/templates/problem/circuitschematic.yaml: patch does not apply
error: patch failed: xmodule/templates/problem/problem_with_hint_in_latex.yaml:17
error: xmodule/templates/problem/problem_with_hint_in_latex.yaml: patch does not apply
hint: Use 'git am --show-current-patch' to see the failed patch
Applying: fix: upgrade problem templates according newer python versions (#31365)
Patch failed at 0001 fix: upgrade problem templates according newer python versions (#31365)
When you have resolved this problem, run "git am --continue".
If you prefer to skip this patch, run "git am --skip" instead.
To restore the original branch and stop patching, run "git am --abort".
The command '/bin/sh -c curl -fsSL https://github.com/openedx/edx-platform/commit/20b93b8b01276edadddfbbb67f15714fddd81c31.patch | git am' returned a non-zero code: 128
Error: Command failed with status 128: docker build -t docker.io/overhangio/openedx:15.1.0-nightly --network host /_fennx/platform/env/build/openedx
```

I'm preparing a pull request to solve the 2 issues above.
If there are any other `tutor-nightly`-specific changes to make, please let me know to include them in the PR too.

FYI @regisb
